conftest.py

```diff
@@ -544,53 +544,58 @@ def cassandra_dir_and_version(config):
+                    mark_names = [m.name for m in item_module[1].pytestmark]
+                    if mark in mark_names:
+                        return True
 
         return False
 
 
-def _is_skippable(item, mark, include_marked, include_other):
-    if has_mark(item, mark):
+def _is_skippable(item, mark, include_marked, include_non_marked, any_class = False):
+    if has_mark(item, mark, any_class):
         if include_marked:
             return False
         else:
             logger.info("SKIP: Skipping %s because it is marked with %s" % (item, mark))
             return True
     else:
-        if include_other:
+        if include_non_marked:
             return False
         else:
             logger.info("SKIP: Skipping %s because it is not marked with %s" % (item, mark))
             return True
 
+def _can_run_resource_intensive_tests(dtest_config, sufficient_resources):
+    return (dtest_config.force_execution_of_resource_intensive_tests
+            or dtest_config.only_resource_intensive_tests
+            or sufficient_resources)
 
-def is_skippable(item,
-                 include_upgrade_tests,
-                 include_non_upgrade_tests,
-                 include_resource_intensive_tests,
-                 include_non_resource_intensive_tests,
-                 include_vnodes_tests,
-                 include_no_vnodes_tests,
-                 include_no_offheap_memtables_tests):
+def is_skippable(item, dtest_config, sufficient_resources):
+    can_run_resource_intensive_tests = _can_run_resource_intensive_tests(dtest_config, sufficient_resources)
 
-    skippable = False
-
-    skippable = skippable or _is_skippable(item, "upgrade_test", include_upgrade_tests, include_non_upgrade_tests)
-    skippable = skippable or _is_skippable(item, "resource_intensive", include_resource_intensive_tests, include_non_resource_intensive_tests)
-    skippable = skippable or _is_skippable(item, "vnodes", include_vnodes_tests, True)
-    skippable = skippable or _is_skippable(item, "no_vnodes", include_no_vnodes_tests, True)
-    skippable = skippable or _is_skippable(item, "no_offheap_memtables", include_no_offheap_memtables_tests, True)
-    skippable = skippable or _is_skippable(item, "depends_driver", False, True)
+    include_upgrade_tests = dtest_config.execute_upgrade_tests or dtest_config.execute_upgrade_tests_only
+    include_non_upgrade_tests = not dtest_config.execute_upgrade_tests_only
+    include_resource_intensive_tests = can_run_resource_intensive_tests and not dtest_config.skip_resource_intensive_tests
+    include_non_resource_intensive_tests = not dtest_config.only_resource_intensive_tests
+    include_vnodes_tests = dtest_config.use_vnodes
+    include_no_vnodes_tests = not dtest_config.use_vnodes
+    include_no_offheap_memtables_tests = not dtest_config.use_off_heap_memtables
 
+    skippable = (_is_skippable(item, "upgrade_test", include_upgrade_tests, include_non_upgrade_tests, True)
```

Comment:
I have no idea after 1st view what is the semantics of "any_class".
We need to find a better name of change the concept.